### ISSUE TITLE
Compatibility with arRPC or similar

### ISSUE THREAD

**Issue by nfp0** [MCMi460/NSO-RPC#133]
Is this application compatible with unofficial Discord RPC implementations such as [arRPC](https://github.com/OpenAsar/arRPC)?
Is the official Discord client running a hard dependency?

And also, is a GUI really necessary?

I ask this because I would love to leave NSO-RPC running 24/7 on my server or on a Docker container, but if a GUI is really necessary that makes it very hard or impossible.

Regardless, thank you for your hard work!

**Comment by HotaruBlaze** [comment]
So just tested arRPC, it worked without any issues, on both ends so on that front its fine.

Theirs a CLI version of NSO-RPC however it hasn't had much of my personal attention...
I've been thinking about how to implement something similar to this myself however theirs some roadblocks, such as RPC. I wasnt aware of arRPC until now.

**However, arRPC still needs some form of webclient or another discord client running to work**

---
So basically, arRPC works out of the box but for docker you're still missing a piece of the puzzle to get this going

**Comment by HotaruBlaze** [comment]
> So just tested arRPC, it worked without any issues, on both ends so on that front its fine.
> 
> Theirs a CLI version of NSO-RPC however it hasn't had much of my personal attention... I've been thinking about how to implement something similar to this myself however theirs some roadblocks, such as RPC. I wasnt aware of arRPC until now.
> 
> **However, arRPC still needs some form of webclient or another discord client running to work**
> 
> So basically, arRPC works out of the box but for docker you're still missing a piece of the puzzle to get this going

So update, I've been able to make this fully working inside of docker, Using NSO-RPC, arRPC and WebCord, i was able to stitch all 3 of them together to make it working fully headless.

Here is my implementation of this: https://github.com/HotaruBlaze/NSO-RPC-Docker

**Comment by nfp0** [comment]
@HotaruBlaze Fantastic news!
I'm going to give that a spin.
